### PR TITLE
fix: wait for, chart value merge issues, clusterissuer

### PR DESCRIPTION
**Pull request by Morriz**
fixes multiple issues:
- merging of chart values now correct with lodash' `merge`
- wait-for init containers removed from tasks as they already do dns retries

### CODE REVIEW

File: src/common/k8s.ts [hunk 1]
```diff
@@ -162,27 +162,32 @@ type WaitTillAvailableOptions = {
...
+  const options = { status: 200, skipSsl: false, ...opts }
   const debug = terminal('waitTillAvailable')
   const retryOptions: Options = {
     retries: 50,
     maxTimeout: 30000,
   }
 
-  const rejectUnauthorized = !options.skipSsl
+  const globalSkipSsl = !env().NODE_TLS_REJECT_UNAUTHORIZED
+  let rejectUnauthorized = !globalSkipSsl
```
Comment (j-zimnowoda) [inline src/common/k8s.ts:174]: let rejectUnauthorized = !!env().NODE_TLS_REJECT_UNAUTHORIZED

File: src/common/k8s.ts [hunk 2]
```diff
@@ -162,27 +162,32 @@ type WaitTillAvailableOptions = {
...
   await retry(async (bail) => {
-    const res = await fetch(url, fetchOptions)
-    if (res.status !== options.status) {
-      debug.warn(`GET ${res.url} ${res.status} ${options.status}`)
-      bail(new Error(`Response status code differs from expected (${options.status}): ${res.status}`))
+    try {
+      const res = await fetch(url, fetchOptions)
+      if (res.status !== options.status) {
+        debug.warn(`GET ${res.url} ${res.status} ${options.status}`)
+        bail(new Error(`Response status code differs from expected (${options.status}): ${res.status}`))
```
Comment (j-zimnowoda) [inline src/common/k8s.ts:186]: We should only `bail` with error if we want to escape retry.
In this case I believe we want to retry. See example: https://github.com/vercel/async-retry#usage

File: versions.yaml [hunk 3]
```diff
@@ -1,3 +1,3 @@
 api: 0.4.85
 console: 0.4.67
-tasks: 0.2.19
+tasks: fix-wait
```
Comment (j-zimnowoda) [inline versions.yaml:3]: Shall we release first tasks?

File: src/common/k8s.ts [hunk 4]
```diff
@@ -162,27 +162,32 @@ type WaitTillAvailableOptions = {
...
   const debug = terminal('waitTillAvailable')
   const retryOptions: Options = {
     retries: 50,
     maxTimeout: 30000,
   }
 
-  const rejectUnauthorized = !options.skipSsl
+  const globalSkipSsl = !env().NODE_TLS_REJECT_UNAUTHORIZED
+  let rejectUnauthorized = !globalSkipSsl
+  if (opts!.skipSsl !== undefined) rejectUnauthorized = !options.skipSsl
```
Comment (j-zimnowoda) [inline src/common/k8s.ts:175]: if .NODE_TLS_REJECT_UNAUTHORIZED is set to 0, then can we still enforce rejection?

**Review by j-zimnowoda — CHANGES_REQUESTED** [review]
Thanks for cleaning up jobs and fixing bootstrap.
There are some questions regarding retry.

**Review by 0-sv — APPROVED** [review]

**Review by bartusz01 — APPROVED** [review]

File: versions.yaml (hunk 3 above)
Comment (Morriz) [reply versions.yaml:3]: of course

File: src/common/k8s.ts (hunk 2 above)
Comment (Morriz) [reply src/common/k8s.ts:186]: it is beneficial to bail early if we know retrying is not possible, which is mostly the case. I excluded 404 and 503 from that so those still throw and retry

File: src/common/k8s.ts (hunk 4 above)
Comment (Morriz) [reply src/common/k8s.ts:175]: I don't assume so...global means node will use that

File: src/common/k8s.ts (hunk 1 above)
Comment (Morriz) [reply src/common/k8s.ts:174]: no, as global is base, but `skipSsl` is explicitly telling it otherwise

**Pull request merged**